### PR TITLE
match paymentToken decimals

### PR DESCRIPTION
**Pull request by RusseII**
This removes all the scaling logic and uses the same decimals between the `bonds` and the `paymentToken`
In a follow up - I'll rename `maxSupply` everywhere to `bonds` in the config. 
There may be some tests that are irrelevant now. In the future I plan on investigating and pruning all the tests.

### CODE REVIEW

File: contracts/Bond.sol [hunk 1]
```diff
@@ -27,6 +28,8 @@ contract Bond is
 
     using FixedPointMathLib for uint256;
 
+    uint8 private _decimals;
```
Comment (Namaskar-1F64F) [inline contracts/Bond.sol:31]: Is there a reason not to be `internal`?

File: contracts/Bond.sol [hunk 2]
```diff
@@ -490,53 +492,26 @@ contract Bond is
...
-    */
-    function _upscale(uint256 amount) internal view returns (uint256) {
-        return amount.mulWadUp(_computeScalingFactor(paymentToken));
+        uint256 amountOverpaid = (paymentBalance()) - totalSupply();
+        return (amountOverpaid);
     }
 
-    function _downscale(uint256 amount) internal view returns (uint256) {
-        return amount.divWadDown(_computeScalingFactor(paymentToken));
+    function decimals() public view override returns (uint8) {
```
Comment (Namaskar-1F64F) [inline contracts/Bond.sol:514]: and this be `external`?

**Review by Namaskar-1F64F — CHANGES_REQUESTED** [review]
All of the things in `Bond` have unnecessary parenthesis around them.

**Comment by RusseII** [comment]
> All of the things in Bond have unnecessary parenthesis around them.

Do you have a regex for me? I was hoping that prettier would remove them :(

File: contracts/Bond.sol (hunk 2 above)
Comment (RusseII) [reply contracts/Bond.sol:514]: image

File: contracts/Bond.sol (hunk 1 above)
Comment (RusseII) [reply contracts/Bond.sol:31]: fixed

Comment (RusseII) [reply contracts/Bond.sol]: Should I even be storing this? Or in decimals should I just return 

```solidity
    function decimals() public view override returns (uint8) {
         return paymentToken.decimals()
}
    ```

**Review by RusseII — COMMENTED** [review]
I think I need to fix the collateral and convertible calculations

**Comment by RusseII** [comment]
> I think I need to fix the collateral and convertible calculations

Actually I think it's fine ... we can still use WAD to store the collateralRatio. https://github.com/porter-finance/v1-core/issues/168#issuecomment-1087996986

**Review by Namaskar-1F64F — APPROVED** [review]
🌋

**Pull request merged**